Ejercicio 1: Área de un triángulo
Escribe un programa en Python que aplique el modelo Entrada-Proceso-Salida para calcular el área de un triángulo utilizando la fórmula:
area = (base * altura) / 2
El programa debe utilizar input(), float() y print(). La entrada debe ser una base de 8 y una altura de 5.
Explicación:
En este ejercicio primero pedimos la base y la altura del triángulo. Después hacemos la operación para obtener el área y finalmente mostramos el resultado en pantalla.

In [1]:
# ==========================================
# EJERCICIO 1: Área de un triángulo
# ==========================================

base = float(input("Ingresa la base del triángulo: "))
altura = float(input("Ingresa la altura del triángulo: "))

area = (base * altura) / 2

print("El área del triángulo es:", area)

El área del triángulo es: 22.5


Ejercicio 2: Conversión de Celsius a Fahrenheit
Escribe un programa en Python que pida una temperatura en grados Celsius y la convierta a grados Fahrenheit utilizando la fórmula:
F = C * 9/5 + 32
La entrada debe ser de 30 grados Celsius.
Explicación:
Aquí convertimos una temperatura de Celsius a Fahrenheit. Primero recibimos el valor con input(), después aplicamos la fórmula y al final mostramos la temperatura convertida.

In [2]:
# ==========================================
# EJERCICIO 2: Celsius a Fahrenheit
# ==========================================

celsius = float(input("Ingresa la temperatura en grados Celsius: "))

fahrenheit = celsius * 9 / 5 + 32

print(celsius, "grados Celsius equivalen a", fahrenheit, "grados Fahrenheit")

49.0 grados Celsius equivalen a 120.2 grados Fahrenheit


Ejercicio 3: Interés simple
Escribe un programa en Python que pida el capital, la tasa de interés y el tiempo, y calcule el interés simple mediante la fórmula:
interes = capital * (tasa / 100) * tiempo
La entrada es un capital de 2000, una tasa de 6 y un tiempo de 2.
Explicación:
En este ejercicio usamos tres datos para calcular cuánto interés se genera. Convertimos la tasa a porcentaje decimal y realizamos la operación para obtener el interés.

In [3]:
# ==========================================
# EJERCICIO 3: Interés simple
# ==========================================

capital = float(input("Ingresa el capital: "))
tasa = float(input("Ingresa la tasa de interés: "))
tiempo = float(input("Ingresa el tiempo: "))

interes = capital * (tasa / 100) * tiempo

print("El interés simple es:", interes)

El interés simple es: 11.249999999999998


Ejercicio 4: Precedencia de operadores
Escribe un programa en Python que calcule la expresión:
4 * 3 + 10 // 3 - 8 % 5
El programa debe mostrar el resultado y el orden de evaluación aplicado:
12 + 3 - 3
Explicación:
En este ejercicio practicamos la prioridad que tienen los operadores matemáticos. Primero se realizan las multiplicaciones, divisiones y módulos, y después las sumas y restas.

In [5]:
# ==========================================
# EJERCICIO 4: Precedencia de operadores
# ==========================================

resultado = 4 * 3 + 10 // 3 - 8 % 5

print("4 * 3 + 10 // 3 - 8 % 5 = 12 + 3 - 3 =", resultado)

4 * 3 + 10 // 3 - 8 % 5 = 12 + 3 - 3 = 12


Ejercicio 5: Casting después de input()
Escribe un programa en Python que pida un número mediante input(), demuestre que el dato recibido es de tipo str, intente realizar x + 1 dentro de un try-except capturando el TypeError y finalmente convierta el dato a entero utilizando int(x) para poder realizar la suma.
Explicación:
Este ejercicio sirve para entender que input() recibe los datos como texto. Por eso no podemos sumar directamente un número entero hasta convertirlo con int().

In [ ]:
# ==========================================
# EJERCICIO 5: Casting después de input()
# ==========================================

x = input("Ingresa un número: ")

print("x es tipo", type(x).__name__)

try:
    resultado = x + 1
except TypeError:
    print("x + 1 genera TypeError")

x = int(x)

print("int(x) + 1 =", x + 1)

Ejercicio 6: Conversión de horas a segundos
Crea un programa que pida una cantidad de horas y la convierta a segundos. Se debe considerar que una hora equivale a 3600 segundos. La entrada es de 3 horas.
Explicación:
Primero recibimos las horas y después las multiplicamos por 3600 para obtener su equivalente en segundos. Finalmente mostramos el resultado.

In [ ]:
# ==========================================
# EJERCICIO 6: Horas a segundos
# ==========================================

horas = float(input("Ingresa la cantidad de horas: "))

segundos = horas * 3600

print(f"{horas:g} horas equivalen a {segundos:g} segundos")

Ejercicio 7: Descuento del 15% con f-strings
Crea un programa que pida el precio de un artículo y aplique un descuento del 15%. Debe mostrar el monto del descuento y el precio final utilizando f-strings y dos decimales. La entrada es de 460.
Explicación:
En este ejercicio calculamos primero cuánto dinero representa el 15% de descuento. Después restamos ese descuento al precio original para obtener el precio final.

In [ ]:
# ==========================================
# EJERCICIO 7: Descuento del 15%
# ==========================================

precio = float(input("Ingresa el precio del artículo: "))

descuento = precio * 0.15
precio_final = precio - descuento

print(f"Descuento (15%): ${descuento:.2f}")
print(f"Precio final: ${precio_final:.2f}")

Ejercicio 8: Número par o impar
Escribe un programa que reciba un número y determine si es par o impar. La entrada es 17 y el resultado esperado es que se indique que es un número impar.
Explicación:
Para saber si un número es par utilizamos el operador %. Si el residuo de dividir entre 2 es 0, es par; si no, es impar.

In [ ]:
# ==========================================
# EJERCICIO 8: Número par o impar
# ==========================================

numero = int(input("Ingresa un número: "))

if numero % 2 == 0:
    print(numero, "es un número par")
else:
    print(numero, "es un número impar")

Ejercicio 9: Mayor de tres números
Escribe un programa que reciba tres números y determine cuál de ellos es el mayor. La entrada es 12, 45 y 23.
Explicación:
Comparamos los tres números utilizando condiciones. Guardamos el número mayor y lo vamos comparando con los otros valores hasta encontrar el mayor.

In [ ]:
# ==========================================
# EJERCICIO 9: Mayor de tres números
# ==========================================

numero1 = float(input("Ingresa el primer número: "))
numero2 = float(input("Ingresa el segundo número: "))
numero3 = float(input("Ingresa el tercer número: "))

mayor = numero1

if numero2 > mayor:
    mayor = numero2

if numero3 > mayor:
    mayor = numero3

print("El mayor de los tres números es:", mayor)

Ejercicio 10: Suma acumulada con while
Pide números mientras sean diferentes de 0. El número 0 debe terminar el ciclo y no debe sumarse. La entrada es 5, 8, 2 y 0.
Explicación:
Usamos un ciclo while para seguir pidiendo números. Cada número se va agregando a una suma y cuando se introduce 0, el ciclo termina.

In [ ]:
# ==========================================
# EJERCICIO 10: Suma acumulada
# ==========================================

suma = 0

numero = int(input("Ingresa un número (0 para terminar): "))

while numero != 0:
    suma = suma + numero
    numero = int(input("Ingresa otro número (0 para terminar): "))

print("La suma de los números es:", suma)

Ejercicio 11: Tabla de multiplicar con for
Crea un programa que pida un número y muestre su tabla de multiplicar del 1 al 10. La entrada es el número 6.
Explicación:
Utilizamos un ciclo for para repetir la multiplicación diez veces. En cada vuelta multiplicamos el número por el valor correspondiente del 1 al 10.

In [ ]:
# ==========================================
# EJERCICIO 11: Tabla de multiplicar
# ==========================================

numero = int(input("Ingresa un número: "))

for i in range(1, 11):
    resultado = numero * i
    print(numero, "x", i, "=", resultado)

Ejercicio 12: Suma de pares del 1 al 20 con for y continue
Crea un programa que recorra los números del 1 al 20, identifique los números pares utilizando un ciclo for y continue, y calcule la suma de todos los números pares.
Explicación:
Recorremos los números del 1 al 20. Cuando encontramos un número impar usamos continue para saltarlo y solamente sumamos los números pares.

In [ ]:
# ==========================================
# EJERCICIO 12: Suma de números pares
# ==========================================

suma = 0

for numero in range(1, 21):

    if numero % 2 != 0:
        continue

    suma = suma + numero

print("La suma de los números pares del 1 al 20 es:", suma)

Ejercicio 13: Suma de una lista con función
Crea una función que reciba una lista y regrese la suma de sus elementos sin utilizar la función sum(). La lista es:
[10, 20, 30, 40]
Explicación:
Creamos una función que recorre todos los elementos de la lista y los va acumulando en una variable. Al terminar, la función regresa el resultado.

In [ ]:
# ==========================================
# EJERCICIO 13: Suma de una lista
# ==========================================

def sumar_lista(lista):
    suma = 0

    for numero in lista:
        suma = suma + numero

    return suma


numeros = [10, 20, 30, 40]

resultado = sumar_lista(numeros)

print("La suma de los elementos es:", resultado)

Ejercicio 14: Máximo de una lista
Crea una función que reciba una lista y regrese el elemento máximo sin utilizar la función max(). La lista es:
[7, 3, 9, 2]
Explicación:
Comenzamos tomando el primer elemento como el mayor. Después recorremos la lista y comparamos los demás elementos para encontrar cuál tiene el valor más grande.

In [ ]:
# ==========================================
# EJERCICIO 14: Máximo de una lista
# ==========================================

def encontrar_maximo(lista):
    mayor = lista[0]

    for numero in lista:
        if numero > mayor:
            mayor = numero

    return mayor


numeros = [7, 3, 9, 2]

resultado = encontrar_maximo(numeros)

print("El elemento máximo es:", resultado)

Ejercicio 15: Matriz y suma de la diagonal principal
Crea una matriz de 3 × 3 con los valores:
[[1,2,3],[4,5,6],[7,8,9]]
Después calcula la suma de los elementos que forman la diagonal principal.
Explicación:
La diagonal principal está formada por los elementos que tienen la misma posición de fila y columna. En este caso son 1, 5 y 9, por lo que la suma es 15.

In [ ]:
# ==========================================
# EJERCICIO 15: Diagonal principal
# ==========================================

matriz = [
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9]
]

suma = 0

for i in range(len(matriz)):
    suma = suma + matriz[i][i]

print("La suma de la diagonal principal es:", suma)

Ejercicio 16: Promedio con función que no regresa valor
Crea una función que reciba una lista de calificaciones y calcule el promedio. La función debe mostrar directamente el resultado y no regresar un valor. La lista es:
[8, 9, 10, 7]
Explicación:
La función suma las calificaciones y las divide entre la cantidad de elementos. Después imprime el promedio utilizando dos decimales.

In [ ]:
# ==========================================
# EJERCICIO 16: Promedio
# ==========================================

def calcular_promedio(calificaciones):
    suma = 0

    for calificacion in calificaciones:
        suma = suma + calificacion

    promedio = suma / len(calificaciones)

    print(f"El promedio es: {promedio:.2f}")


calificaciones = [8, 9, 10, 7]

calcular_promedio(calificaciones)

Ejercicio 17: Invertir una lista con slicing
Crea un programa que utilice slicing para invertir la siguiente lista:
[1, 2, 3, 4, 5]
El resultado debe ser:
[5, 4, 3, 2, 1]
Explicación:
Utilizamos [::-1] para recorrer la lista desde el último elemento hasta el primero. De esta manera podemos invertirla fácilmente.

In [ ]:
# ==========================================
# EJERCICIO 17: Lista invertida
# ==========================================

numeros = [1, 2, 3, 4, 5]

lista_invertida = numeros[::-1]

print("Lista invertida:", lista_invertida)

Ejercicio 18: Factorial con función recursiva
Crea una función recursiva que calcule el factorial de un número. La entrada es 5 y el resultado esperado es 120.
Explicación:
Una función recursiva es una función que se llama a sí misma. En este caso va multiplicando el número por el factorial del número anterior hasta llegar a 1.

In [ ]:
# ==========================================
# EJERCICIO 18: Factorial recursivo
# ==========================================

def factorial(numero):

    if numero == 0 or numero == 1:
        return 1

    return numero * factorial(numero - 1)


numero = int(input("Ingresa un número: "))

resultado = factorial(numero)

print("El factorial de", numero, "es:", resultado)

Ejercicio 19: Tupla: acceso y suma con función
Utiliza la siguiente tupla:
(4, 8, 15, 16, 23, 42)
El programa debe mostrar el tercer elemento y calcular la suma de todos los elementos mediante una función.
Explicación:
En este ejercicio practicamos el uso de tuplas. Para obtener el tercer elemento utilizamos el índice 2, ya que Python comienza a contar desde cero. También hacemos una función para sumar sus elementos.

In [ ]:
# ==========================================
# EJERCICIO 19: Tupla
# ==========================================

def sumar_tupla(tupla):
    suma = 0

    for numero in tupla:
        suma = suma + numero

    return suma


numeros = (4, 8, 15, 16, 23, 42)

print("El tercer elemento es:", numeros[2])

resultado = sumar_tupla(numeros)

print("La suma de la tupla es:", resultado)

Ejercicio 20: Diccionario: buscar teléfono de un contacto
Crea un diccionario de contactos que contenga números telefónicos. Después pide el nombre de un contacto y busca su teléfono. La entrada es Beto y el teléfono esperado es 5555678.
Explicación:
Utilizamos un diccionario para guardar los nombres y teléfonos. Después buscamos el nombre que introdujo el usuario y mostramos el teléfono correspondiente.

In [ ]:
# ==========================================
# EJERCICIO 20: Teléfono de un contacto
# ==========================================

contactos = {
    "Ana": "5551234",
    "Beto": "5555678",
    "Carlos": "5559876"
}

nombre = input("Ingresa el nombre del contacto: ")

if nombre in contactos:
    print("El teléfono de", nombre, "es:", contactos[nombre])
else:
    print("El contacto no existe")

Ejercicio 21: Diccionario de calificaciones y promedios
Crea un diccionario que contenga las calificaciones de Ana y Beto:
{"Ana": [8, 9, 10], "Beto": [7, 7, 8]}
El programa debe calcular y mostrar el promedio de cada estudiante.
Explicación:
Cada nombre tiene una lista con sus calificaciones. Recorremos el diccionario y calculamos el promedio de cada lista para mostrarlo con dos decimales.

In [ ]:
# ==========================================
# EJERCICIO 21: Calificaciones y promedios
# ==========================================

calificaciones = {
    "Ana": [8, 9, 10],
    "Beto": [7, 7, 8]
}

for nombre, notas in calificaciones.items():

    suma = 0

    for nota in notas:
        suma = suma + nota

    promedio = suma / len(notas)

    print(f"{nombre}: {promedio:.2f}")

Ejercicio 22: Excepción de división entre cero
Crea un programa que realice una división y utilice una excepción para controlar el caso en que el segundo número sea cero. Para la entrada 10 y 0 debe mostrar:
Error: No se puede dividir entre cero.
Explicación:
Utilizamos try-except para evitar que el programa termine con un error cuando intentamos dividir entre cero. En lugar de detenerse, mostramos un mensaje al usuario.

In [ ]:
# ==========================================
# EJERCICIO 22: División entre cero
# ==========================================

try:
    numero1 = int(input("Ingresa el primer número: "))
    numero2 = int(input("Ingresa el segundo número: "))

    resultado = numero1 / numero2

    print("El resultado es:", resultado)

except ZeroDivisionError:
    print("Error: No se puede dividir entre cero.")

Ejercicio 23: Excepción de valor no numérico
Crea un programa que intente convertir una entrada a número entero y controle el error cuando el usuario introduzca un valor que no sea numérico. La entrada es "abc".
Explicación:
Usamos try-except para controlar el ValueError. Si el usuario escribe letras en lugar de un número entero, el programa muestra un mensaje indicando que debe ingresar un número.

In [ ]:
# ==========================================
# EJERCICIO 23: Excepción de valor no numérico
# ==========================================

try:
    numero = int(input("Ingresa un número entero: "))

    print("El número ingresado es:", numero)

except ValueError:
    print("Error: Debes ingresar un número entero.")

Ejercicio 24: Contar palabras de un string
Crea un programa que reciba el siguiente mensaje:
"Python es un lenguaje de programacion"
Después debe contar cuántas palabras contiene y mostrar el resultado.
Explicación:
Utilizamos split() para separar el texto en palabras. Después usamos len() para saber cuántas palabras hay en la lista obtenida.

In [ ]:
# ==========================================
# EJERCICIO 24: Contar palabras
# ==========================================

mensaje = input("Ingresa un mensaje: ")

palabras = mensaje.split()

cantidad = len(palabras)

print("El mensaje tiene", cantidad, "palabras.")

Ejercicio 25: Longitud, mayúsculas y reemplazo de un string
Crea un programa que reciba el texto:
"aprende python"
Después debe mostrar su longitud, convertirlo a mayúsculas y reemplazar la palabra "python" por "programacion".
Explicación:
En este ejercicio practicamos diferentes operaciones con strings. Usamos len() para conocer la longitud, .upper() para convertir el texto a mayúsculas y .replace() para cambiar una palabra por otra.

In [ ]:
# Ejercicio 25: Longitud, mayúsculas y reemplazo de un string

# Se guarda el texto en una variable
mensaje = "aprende python"

# Se muestra la longitud del texto
print("Longitud:", len(mensaje))

# Se convierte el texto a mayúsculas
print("En mayúsculas:", mensaje.upper())

# Se reemplaza la palabra python por programacion
print("Reemplazo:", mensaje.replace("python", "programacion"))